src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+            return parseCatch(node, checker, synthNamesMap, lastDotThenMap, context, prevArgName);
+        }
+        else if (isPropertyAccessExpression(node)) {
+            return parseCallback(node.expression, checker, outermostParent, synthNamesMap, lastDotThenMap, context, prevArgName);
+        }
+        else if (returnsAPromise(node, checker)) {
+            return parsePromiseCall(node, lastDotThenMap, prevArgName);
+        }
+
+        return [];
+    }
+
+    function parseCatch(node: CallExpression, checker: TypeChecker, synthNamesMap: Map<[Identifier, number]>, lastDotThenMap: Map<boolean>, context: CodeFixContextBase, prevArgName?: [Identifier, number]): Statement[] {
+        const func = node.arguments[0];
+        const argName = getArgName(func, synthNamesMap, checker);
+
+        let varDecl;
+        if (prevArgName && lastDotThenMap.get(String(getNodeId(node)))) {
+            varDecl = createVariableStatement(/*modifiers*/ undefined, createVariableDeclarationList([createVariableDeclaration(getSynthesizedDeepClone(prevArgName[0]))], NodeFlags.Let));
+            prevArgName[1] += 2;
+        }
+        const tryBlock = createBlock(parseCallback(node.expression, checker, node, synthNamesMap, lastDotThenMap, context, prevArgName));
+
+        const callbackBody = getCallbackBody(func, prevArgName, argName, node, checker, synthNamesMap, lastDotThenMap, context);
+        const catchClause = createCatchClause(argName[0].text, createBlock(callbackBody));
+
+        const tryStatement = createTry(tryBlock, catchClause, /*finallyBlock*/ undefined);
+        return varDecl ? [varDecl, tryStatement] : [tryStatement];
+    }
+
+    function parseThen(node: CallExpression, checker: TypeChecker, outermostParent: CallExpression, synthNamesMap: Map<[Identifier, number]>, lastDotThenMap: Map<boolean>, context: CodeFixContextBase, prevArgName?: [Identifier, number]): Statement[] {
+        const [res, rej] = node.arguments;
+
+        if (!res) {
+            return parseCallback(node.expression, checker, outermostParent, synthNamesMap, lastDotThenMap, context, prevArgName);
+        }
+
+        const argNameRes = getArgName(res, synthNamesMap, checker);
+        const callbackBody = getCallbackBody(res, prevArgName, argNameRes, node, checker, synthNamesMap, lastDotThenMap, context);
+
+        if (rej) {
+            const argNameRej = getArgName(rej, synthNamesMap, checker);
+
+            const tryBlock = createBlock(parseCallback(node.expression, checker, node, synthNamesMap, lastDotThenMap, context, argNameRes).concat(callbackBody));
+
+            const callbackBody2 = getCallbackBody(rej, prevArgName, argNameRej, node, checker, synthNamesMap, lastDotThenMap, context);
+            const catchClause = createCatchClause(argNameRej[0].text, createBlock(callbackBody2));
+
+            return [createTry(tryBlock, catchClause, /*finallyBlock*/ undefined) as Statement];
+        }
+        else if (res) {

```

Comment:
`res` is definitely truthy here (we returned earlier in the function if it wasn't)